src/AbstractOptions.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Equip\Queue;
+
+use Equip\Command\OptionsInterface;
+use Equip\Command\OptionsSerializerTrait;
+use Equip\Queue\Exception\MessageException;
+
+abstract class AbstractOptions implements OptionsInterface
```

Comment:
Is there a need for this to be `abstract` ? We could add a constructor and accept `$command` and `$queue` values as parameters to it. I realize the advantage of subclassing is for typehinting, but it would still be possible to use that approach.